Makefile

```diff
@@ -254,7 +254,7 @@ ifeq (, $(shell which controller-gen))
 	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$CONTROLLER_GEN_TMP_DIR ;\
```

Comment:
It turned out that when I ran make ctrl-generate and make ctrl-manifests probably one or both of those commands errored because I couldn't use the CRD when installing. I found via github issues that older versions of controller-gen can panic so bumped it here

Reply:
Sounds reasonable to bump this. ❓ when was the fix introduced? It looks like maybe `controller-gen` is intended for compatibility with a specific k8s.io version, so we may want to either upgrade to an earlier fix version, or bump our k8s.io dependencies as well: https://github.com/kubernetes-sigs/controller-tools/tree/main?tab=readme-ov-file#compatibility

(I’d lean toward bumping k8s.io but that could lead to a cascading bump of `controller-runtime` and other deps, which we’d ideally do with ample time to vet them)

Reply:
I see evidence in the convo here that controller-gen 0.14 could work, rather than bumping all the way to 0.16, do you think that's preferable, and we reduce the risk of things being super out of date?

Reply:
Yeah I feel like the lowest viable / other-deps-aligned bump probably reduces risk now, then a fast follow ticket to get our dependencies updated fully in the next release (aligning w/ our K8s 1.30 support). WDYT?

Reply:
Yup that makes sense!